### ISSUE TITLE
Replace celluloid with thin actor model based on ruby Threads?

### ISSUE THREAD

**Issue by lamont-granquist** [berkshelf/ridley#361]
Lots of bugs seem to be celluloid-based, but the ridley use of celluloid does not seem to be terribly complicated.

**Comment by janko-m** [comment]
I agree. At the moment all exceptions that happen during cookbook upload are raised as an obscure 
`task was terminated (Celluloid::Task::TerminatedError)`.
